ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/db/ColumnIdCachingDasDbTest.java

```diff
@@ -99,9 +101,50 @@ void checkCacheIsInvalidated() {
     assertThat(res2).isCompletedWithValueMatching(l -> !l.isEmpty());
   }
 
+  @Test
+  void checkWriteCacheIsUsed() {
+    SafeFuture<List<DataColumnSlotAndIdentifier>> res1 =
+        columnIdCachingDb.getColumnIdentifiers(UInt64.valueOf(777));
+
+    stubAsync.advanceTimeGradually(ofMillis(1));
+
+    final DataColumnSidecar sidecar = createSidecar(777, 77);
+    SafeFuture<Void> addCompleteFuture1 = columnIdCachingDb.addSidecar(sidecar);
+    stubAsync.advanceTimeGraduallyUntilAllDone(ofSeconds(1));
+
+    assertThat(res1)
+        .isCompleted(); // no assumptions on result: cache may or may not pick up latest changes
+    assertThat(addCompleteFuture1).isCompleted();
+    long reads1 = db.getDbReadCounter().get();
```

Comment:
we either need a lot of finals or to check these in the asserts

Reply:
My bad, used to do it in hacky PoC style. Right time to get better. Updated.